kbfsgit/runner.go

```diff
@@ -629,6 +633,28 @@ func (r *runner) printJournalStatusUntilFlushed(
 
 const unlockPrintBytesStatusThreshold = time.Second / 2
 
+func (r *runner) printDoneIfNeeded(elapsedStr string) {
+	r.needPrintDoneLock.Lock()
+	defer r.needPrintDoneLock.Unlock()
+	// go-git grabs the lock right after plumbing.StatusIndexOffset, but before
+	// sending the Done status update. As a result, it would look like we are
+	// flushing the journal before plumbing.StatusIndexOffset is done. So
+	// instead, print "done." only if it's not printed yet.
+	if r.needPrintDone {
+		r.errput.Write([]byte("done." + elapsedStr + "\n"))
+		r.needPrintDone = false
+	}
+}
+
+func (r *runner) needToPrintDone(toPrint []byte) {
```

Comment:
This function name is confusing, because 1) it sort of sounds like a question, rather than a command, and 2) it prints a string in addition to setting the `needPrintDone` bool.

How about `printStageStart`?  (And maybe `printStageEnd` for the function above, for the sake of consistency?)

Reply:
Sounds good!